### PR TITLE
Trip image background

### PR DESCRIPTION
**Pull request by eleonoraka**
## Summary

I added a new feature which is the trip image. The user can select an image from their device and set it as trip image. If the user doesn't select an image, a default image is provided. I consequently updated all the files related to the trips

## Changes

- **Models:** added contribute imageUri to Trip data class; added function to upload image to Firebase in TripsViewModel
- **Screens/UI:** added trip image to AddTrip screen; added preview trip image to Overview screen

##  Testing

Explain how this feature was tested and what kind of testing (unit, integration, UI) has been done. Mention platforms or configurations used for testing.
- **Manual Testing:**
    - [x] Tested on emulator / physical device.
- **Unit Tests**
    - [ ] Added units tests for this

Remark: waiting to merge to add tests

### CODE REVIEW

File: app/src/main/java/com/android/voyageur/ui/overview/Overview.kt [hunk 1]
```diff
@@ -85,11 +92,25 @@ fun TripItem(trip: Trip) {
...
-            textAlign = TextAlign.Center)
-      },
-  )
+        Row {
+          if (trip.imageUri.isNotEmpty()) {
+            Image(
+                painter = rememberAsyncImagePainter(model = trip.imageUri),
+                contentDescription = "Selected image",
+                contentScale = ContentScale.Crop,
+                modifier = Modifier.width(120.dp).height(217.dp))
```
Comment (ioanajianu) [inline app/src/main/java/com/android/voyageur/ui/overview/Overview.kt:101]: You don't necessarily have to address this comment now but,
These fixed values for width and height made me think about what would happen on larger/smaller screens, maybe the image would appear too big/too small. 
Not sure if this is a singular case, we could have used fixed values in other files as well, so we should test the whole app on other device resolutions as well. I'll add this as a task on the Scrum Board, it's probably not doable until M1 but we should consider it for future Sprints.

**Review by ioanajianu — CHANGES_REQUESTED** [review]
Looks good, the "Select Image from Gallery" feature is a nice touch :)
Just a few things about tests:
* Could you please add tests for this change in `OverviewScreenTest`? Just something that checks that an image is displayed for a trip.
* And regarding the changes in `AddTrip`, discuss with Mircea about the tests, he was working on `AddTripTests`.

About the comment I left you about the image width and height, you decide if there's time to address it now or we can address it in the future when we test the whole app functionality on other screen resolutions.

File: app/src/main/java/com/android/voyageur/ui/overview/Overview.kt (hunk 1 above)
Comment (apana03) [reply app/src/main/java/com/android/voyageur/ui/overview/Overview.kt:101]: +1

File: app/src/main/java/com/android/voyageur/model/trip/TripsViewModel.kt [hunk 2]
```diff
@@ -75,4 +77,19 @@ class TripsViewModel(
...
+        .putFile(uri)
+        .addOnSuccessListener {
+          fileRef.downloadUrl
+              .addOnSuccessListener { downloadUri ->
+                onSuccess(downloadUri.toString()) // Return the download URL
+              }
+              .addOnFailureListener { exception -> onFailure(exception) }
+        }
+        .addOnFailureListener { exception -> onFailure(exception) }
+  }
```
[the remark this answers is not in the document]
Comment (apana03) [reply app/src/main/java/com/android/voyageur/model/trip/TripsViewModel.kt:94]: +1

File: app/src/main/java/com/android/voyageur/ui/overview/Overview.kt (hunk 1 above)
Comment (eleonoraka) [reply app/src/main/java/com/android/voyageur/ui/overview/Overview.kt:101]: For this PR let's prioritize merging the basic functionalities for the 1st milestone. We can address this in a future PR.

File: app/src/main/java/com/android/voyageur/ui/overview/Overview.kt [hunk 3]
```diff
@@ -85,11 +92,25 @@ fun TripItem(trip: Trip) {
...
-            text = trip.name,
-            modifier = Modifier.padding(16.dp),
-            fontSize = 20.sp,
-            textAlign = TextAlign.Center)
-      },
-  )
+        Row {
+          if (trip.imageUri.isNotEmpty()) {
+            Image(
+                painter = rememberAsyncImagePainter(model = trip.imageUri),
```
[the remark this answers is not in the document]
Comment (eleonoraka) [reply app/src/main/java/com/android/voyageur/ui/overview/Overview.kt:98]: For this PR let's prioritize merging the basic functionalities for the 1st milestone. We can address this in a future PR.

**Comment by eleonoraka** [comment]
>     * Could you please add tests for this change in `OverviewScreenTest`? Just something that checks that an image is displayed for a trip.

I am waiting for the `OverviewScreenTest` to be uploaded and check how I can integrate these tests in the file.

File: app/src/main/java/com/android/voyageur/model/trip/TripsViewModel.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (eleonoraka) [reply app/src/main/java/com/android/voyageur/model/trip/TripsViewModel.kt:94]: I agree with your remarks. For this PR let's prioritize merging the basic functionalities for the 1st milestone. I will address this issue as soon as possible before the 1st milestone deadline.

**Review by ioanajianu — APPROVED** [review]
LGTM, thanks for adding the tests

**Pull request merged**